src/aipdf/ocr.py

```
@@ -71,6 +104,56 @@ def image_to_markdown(file_object, client, model="gpt-4o",  prompt=DEFAULT_PROMP
     except Exception as e:
         logging.error(f"An error occurred while processing the image: {e}")
         return None
+    
+
+async def image_to_markdown_async(file_object, client, model="gpt-4o", prompt=DEFAULT_PROMPT):
+    """
+    Asynchronously process a single image file and convert its content to markdown using OpenAI's API.
+
+    Args:
+        file_object (io.BytesIO): The image file object.
+        client (AsyncOpenAI): The AsyncOpenAI client instance.
+        model (str, optional): by default is gpt-4o
+        prompt (str, optional): The prompt to send to the API. Defaults to DEFAULT_PROMPT.
+
+    Returns:
+        tuple: A tuple containing the page number and the markdown representation of the image content, or None if an error occurs.
+    """
+    # Log that we're about to process a page
+    logging.info("About to process a page")
+
+    base64_image = base64.b64encode(file_object.read()).decode('utf-8')
+
+    try:
+        response = await client.chat.completions.create(
+            model=model,
+            messages=[
+                {
+                    "role": "user",
+                    "content": [
+                        {
+                            "type": "text",
+                            "text": prompt
+                        },
+                        {
+                            "type": "image_url",
+                            "image_url": {
+                                "url": f"data:image/jpeg;base64,{base64_image}"
+                            }
+                        }
+                    ]
+                }
+            ]
+        )
+
+        # Extract the markdown content from the response
+        markdown_content = response.choices[0].message.content
+        logging.info("Page processed successfully")
+        return markdown_content
+
+    except Exception as e:
```

what the reason to hide error?

---

@ea-rus Since we could be processing multiple pages here; the error is not raised for a single request because the other API calls could potentially succeed.